### ISSUE TITLE
Segfault on exit

### ISSUE THREAD

**Issue by oxillo** [openframeworks/openFrameworks#5096]
On MSYS2, segfault happens when exiting with the esc key.

**Comment by danzeeeman** [comment]
Does this happen on all app? I've seen some segfaults on master when
closing apps with multiple threads and when they aren't stopped correctly
on OSX.

Can you document this bug a bit more with stack traces etc and steps to
reproduce?

Thanks!

On Saturday, May 21, 2016, Olivier XILLO <notifications@github.com> wrote:

> On MSYS2, segfault happens when exiting with the esc key.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openframeworks/openFrameworks/issues/5096>
>


-- 
Sent from Gmail Mobile #autocorrect

**Comment by oxillo** [comment]
Have it on several examples (may be all ?). Commenting the glfwHideWindow call in ofAppGlfwWindow::close() seems to solve the problem. But I need to dig deeper...

**Comment by oxillo** [comment]
I suspect the glfw calbacks to be the reason. 
Disabling them solves the problem.
Will PR that tomorrow after more checks.
 
```C++
void ofAppGLFWWindow::close(){
	if(windowP){
                glfwSetMouseButtonCallback(windowP, nullptr);
                glfwSetCursorPosCallback(windowP, nullptr);
                glfwSetCursorEnterCallback(windowP, nullptr);
                glfwSetKeyCallback(windowP, nullptr);
                glfwSetWindowSizeCallback(windowP, nullptr);
                glfwSetWindowCloseCallback(windowP, nullptr);
                glfwSetScrollCallback(windowP, nullptr);
                glfwSetDropCallback(windowP, nullptr);
		//hide the window before we destroy it stops a flicker on OS X on exit.
		glfwHideWindow(windowP);
		glfwDestroyWindow(windowP);
		windowP = nullptr;
		events().disable();
		bWindowNeedsShowing = true;
	}
}
```

**Comment by tgfrerer** [comment]
I believe that's fixed now that #5098 has been merged.
